src/app/components/core/control-dispatcher/api-flow/add-edit-api-flow/add-edit-api-flow.component.ts

```diff
@@ -0,0 +1,512 @@
+            this.alertService.success(data.message, true);
+            this.router.navigate(['control-dispatcher/api']);
+          },
+          error => {
+            /** request completed but error */
+            this.ngProgress.done();
+            if (error.status === 0) {
+              console.log('service down ', error);
+            } else {
+              this.alertService.error(error.statusText);
+            }
+          });
+    }
+
+    checkAndRequestAPIFlow() {
+      // TODO: FOGL-8079 (blank values for variables are not allowed)
+      return;
+      // if (this.getFormControls('variables').length > 0) {
+      //   return;
+      // }
+      // this.requestAPIFlow();
+    }
+  
+    requestAPIFlow(value = null) {
+      let variables = {};
+      if (value.variables) {       
+        value.variables.forEach(v => { variables[v.vName] = v.vValue });
+      }        
+      this.controlAPIFlowService.requestAPIFlow(this.af.name, variables) 
+      .subscribe((data: any) => {
+          /** request completed */
+          this.ngProgress.done();
+          this.alertService.success(data.message, true);
+          this.closeModal('confirmation-execute-dialog');
+          this.getAPIFlow();
+          },
+          error => {
+          /** request completed but error */
+          this.ngProgress.done();
+          if (error.status === 0) {
+              console.log('service down ', error);
+          } else {
+              this.alertService.error(error.statusText);
+          }
+      });
+    }
+
+    changeType(value) {
+        this.selectedType = value === 'Select Type' ? '' : value;
+    }
+
+    getDestTypes() {
+        this.controlPipelinesService.getSourceDestinationTypeList('destination')
+          .subscribe((data: any) => {
+            this.ngProgress.done();
+            // this.destinationTypes = data;
+            this.destinationTypes = data.filter(d => d.name !== 'Any');
+          }, error => {
+            if (error.status === 0) {
+              console.log('service down ', error);
+            } else {
+              this.alertService.error(error.statusText);
+            }
+          });
+    }
+
+    navigateToList() {
+        this.router.navigate(['control-dispatcher/api']);
+    }
+
+    onCancel() {
+        this.router.navigate(['control-dispatcher/api']);
+    }
+
+    onCheckboxClicked(event) {
+        this.af.anonymous = event.target.checked;
+    }
+
+    getFormControls(type): AbstractControl[] {
+      return (<FormArray>this.apiFlowForm.get(type)).controls;
+    }
+
+    selectDestinationName(value) {
+        this.selectedDestinationName = value === 'Select Destination Name' ? null : value;
+    }
+
+    getDestinationNames(selectedType) {
+        this.destinationNames = [];
+        this.selectedDestinationName = null;
+        this.af.destination = selectedType.name === 'Select Destination Type' ? '' : selectedType.name;
+        switch (selectedType.name) {
+          case 'Any':
```

Comment:
remove Any; 

Though suggest to use existing  function in control pipeline or move such to common /cc @monika-Dianomic